Jenkinsfile

```diff
@@ -0,0 +1,126 @@
+#!groovy
+
+// Jenkinsfile for compiling, testing, and packaging Aseba.
+// Requires CMake plugin from https://github.com/davidjsherman/aseba-jenkins.git in global library.
+
+// Ideally we should unarchive previously compiled Dashel and Enki libraries, but for
+// now we conservatively recompile them.
+
+pipeline {
+	agent label:'' // use any available Jenkins agent
+
+	// Jenkins will prompt for parameters when a branch is build manually
+	// but will use default parameters when the entire project is built.
+	parameters {
+        stringParam(defaultValue: 'master', description: 'Dashel branch', name: 'branch_dashel')
+        stringParam(defaultValue: 'master', description: 'Enki branch', name: 'branch_enki')
+		booleanParam(defaultValue: false, description: 'Build packages', name: 'do_packaging')
+	}
+	
+	stages {
+		stage('Prepare') {
+			steps {
+				echo "branch_dashel=${env.branch_dashel}"
+				echo "branch_enki=${env.branch_enki}"
+				echo "do_packaging=${env.do_packaging}"
+
+				// Dashel and Enki will be checked ou into externals/ directory.
+				// Everything will be built in build/ directory.
+				// Everything will be installed in dist/ directory.
+				sh 'mkdir -p externals build dist'
+				dir('aseba') {
+					checkout scm
+					sh 'git submodule update --init'
+				}
+				dir('externals/dashel') {
+					git branch: "${env.branch_dashel}", url: 'https://github.com/aseba-community/dashel.git'
+				}
+				dir('externals/enki') {
+					git branch: "${env.branch_enki}", url: 'https://github.com/enki-community/enki.git'
+				}
+				stash excludes: '.git', name: 'source'
+			}
+		}
+		stage('Dashel') {
+			steps {
+				unstash 'source'
+				CMake([buildType: 'Debug',
```

Comment:
I am assuming that there will be a separate Jenkinsfile for packaging.
It could be in Aseba, or in a separate repository for bespoke packages that need special resources like examples or lesson plans. Jenkins can be configured either way.

The Jenkins file should work ‘out of the box’, we should try to minimize parameters that require user intervention.